lstchain/reco/dl1_to_dl2.py

```diff
@@ -768,12 +768,16 @@ def get_expected_source_pos(data, data_type, config, focal_length=28 * u.m):
             expected_src_pos_y_m = np.zeros(len(data))
 
         # compute source position in camera coordinate event by event for wobble mode
-        if config.get('observation_mode') == 'wobble':
+        elif config.get('observation_mode') == 'wobble':
 
             if 'source_name' in config:
                 source_coord = SkyCoord.from_name(config.get('source_name'))
-            else:
+            elif 'source_ra' and 'source_dec' in config:
                 source_coord = SkyCoord(config.get('source_ra'), config.get('source_dec'), frame="icrs", unit="deg")
+            else:
+                raise NameError(
```

Comment:
NameError seems not to be the correct thing to do here... 

KeyError seems more appropriate

Reply:
It seems I implemented it locally at that time, but didn't push it to here... now it was done!